portal/static/js/admin.js

```diff
@@ -1,652 +1,781 @@
-              content += "<TH>" + i18next.t("Type") + "<TH>"+ i18next.t("Report Name") + "</TH><TH>" + i18next.t("Generated (GMT)") + "</TH><TH>" + i18next.t("Downloaded") + "</TH>";
-              documents.forEach(function(item) {
 
-                  var c = item["contributor"];
+  AdminTool.prototype.getReportModal = function(patientId) {
 
-                  /*
-                   * only draw the most recent, same report won't be displayed
-                   */
-                  if (!existingItems[c] && hasValue(c)) {
-                    content += "<tr>" +
-                              "<td>" + c + "</td>" +
-                              "<td>" + item["filename"] + "</td>" +
-                              "<td>" + tnthDates.formatDateString(item["uploaded_at"], "iso") + "</td>" +
-                              "<td class='text-center'>" + '<a title="' + i18next.t("Download") + '" href="' + '/api/user/' + String(item["user_id"]) + '/user_documents/' + String(item["id"])+ '"><i class="fa fa-download"></i></a>' + "</td>"
-                              "</tr>";
-                    existingItems[c] = true;
-                    count++;
-                  };
-              });
-              content += "</table>";
-              content += "<br/>";
-              content += "<a class='btn btn-tnth-primary btn-sm btn-all'>" + i18next.t("View All") + "</a>";
+    $("#patientReportModal").modal("show");
+    $("#patientReportLoader").removeClass("tnth-hide");
 
-              $("#patientReportContent").html(content);
-              if (count > 1) $("#patientReportModal .modal-title").text(i18next.t("Patient Reports"));
-              else $("#patientReportModal .modal-title").text(i18next.t("Patient Report"));
-              $("#patientReportContent .btn-all").attr("href", "patient_profile/"+patientId+"#profilePatientReportTable");
+    var self = this;
+    var tnthDates = self.getDependency("tnthDates"), tnthAjax = self.getDependency("tnthAjax"), i18next = self.getDependency("i18next");
 
-            } else {
-              $("#patientReportMessage").html(i18next("No report data found."));
-            };
+    tnthAjax.patientReport(patientId, function(data) {
+        if (data) {
+          if (!data.error) {
+              if (data["user_documents"] && data["user_documents"].length > 0) {
+                var existingItems = {}, count = 0;
+                /*
+                 * sort to get the latest first
+                 */
+                var documents = data["user_documents"].sort(function(a,b){
+                   return new Date(b.uploaded_at) - new Date(a.uploaded_at);
+                });
+                var content = "<table class='table-bordered table-condensed table-responsive tnth-table'>";
+                content += "<TH>" + i18next.t("Type") + "<TH>"+ i18next.t("Report Name") + "</TH><TH>" + i18next.t("Generated (GMT)") + "</TH><TH>" + i18next.t("Downloaded") + "</TH>";
```

Comment:
HTML/content is easier to read and maintain if generated declarative, like with a template (instead of string concatenation).

I don't think we're including any libs that could help here (mustache), but it might be worth a second look if we start using one.